metals/src/main/scala/scala/meta/internal/metals/PackageProvider.scala

```diff
@@ -0,0 +1,48 @@
+package scala.meta.internal.metals
+
+import org.eclipse.lsp4j.Position
+import org.eclipse.lsp4j.Range
+import org.eclipse.lsp4j.TextEdit
+import org.eclipse.lsp4j.WorkspaceEdit
+
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.io.AbsolutePath
+
+class PackageProvider(private val buildTargets: BuildTargets) {
+
+  def workspaceEdit(path: AbsolutePath): Option[WorkspaceEdit] = {
+    packageName(path).map(workspaceEdit(path, _))
+  }
+
+  private def packageName(path: AbsolutePath): Option[String] = {
+    // package.scala file needs to be handled separately
+    // as its package name is different than package name of normal scala file
+    if (path.isScala &&
+      path.toFile.length() == 0 &&
```

Comment:
why not move it to `path.isEmpty` or `path.isEmptyFile` in extensions?

Also, we can extract a part of the condition and get a oneliner:
`if(isEmptyScalaFile(path) && path.filename != "pacakge.scala")`

Reply:
Might change this in a follow up, but merging now, since everything is green.